### ISSUE TITLE
How to use scripts in the Modal box?

### ISSUE THREAD

**Issue by vanile** [reactjs/react-modal#546]
### Summary:
I'm pretty new to React btw, but I can't figure this out.
Scripts work on the React component when it's not used in a Modal box but when I create the Modal box with the component inside, the scripts no longer work. The styling still works but no scripts. All of my CSS and scripts are sourced in my index.html file.

### Steps to reproduce:

So for my codesandbox project, click 'Add an event' in the top right and a Modal box with a form pops out (I didn't upload my personal styling so it looks kind of weird, but it's not needed). The input elements are material style so when you click on an input, then the label should float upward, not stay still etc. Also the Date and Time pickers no longer work even though they did before when not opened with a Modal box. Another obvious offender is the drop down selector doesn't have the animations from the scripts. So, I guess scripts don't work even though styles do.

### Expected behavior:

TLDR: Need material functionality and animations (works when I use the Form component outside of Modal box but not inside), but not working with scripts. CSS works.

### Link to example of issue:

https://codesandbox.io/s/4q3z65jvx7

### Additional notes:

Thanks

E. Oh the Date and Time pickers won't work in the link because it also needs a local script with the initializers that I made, but you get the point.

**Comment by sw-yx** [comment]
thanks for linking the example. what is this `externals` thing here?
![image](https://user-images.githubusercontent.com/6764957/32149737-71149154-bcdf-11e7-9b79-eb978464a2f7.png)

in general its not a great idea to mix scripts and react like this. have you tried `material-ui`? <http://www.material-ui.com/#/>

**Comment by vanile** [comment]
I am not seeing that in my console for some reason.

I haven't looked at material-ui yet. What I want is to be able to pull up a form box inside the page and not link to another page for the form. I assume I would be able to do that with the Dialog component?

**Comment by sw-yx** [comment]
depends on your needs (im more of a react-bootstrap guy myself). more importantly you can use their inbuilt form components to achieve the move-up effect you are trying to do here

**Comment by diasbruno** [comment]
@vanile Prefer following the material-ui guide and install it on your project (as @sw-yx already explained) to avoid the complexity of loading scripts.

With material-ui installed, you can choose to use react-modal or the dialog that the library provides.

This will give you a better result.

**Comment by diasbruno** [comment]
The thing is when using "non-react" libraries, they may not initialize properly because some elements are not yet available to be queried by the script.

**Comment by vanile** [comment]
Alright, thanks friends. I've taken your advice on using material-ui and it seems to be working out much better.

**Comment by diasbruno** [comment]
Glad to help. Closing this issue.
